Add PCI devices drivers and firmwares details for VMhosts

equelin opened the pull request:
Ideas come from issue #19

<!--- Provide a general summary of your changes in the Title above -->

## Description
Add PCI devices driver and firmware details for each host.

## Modifications
Add a new function named `Get-PCIdeviceDetail' whose role is to gather data about PCI devices, and more specifically the related drivers, firmwares and VIB.

Note: The code is heavily based on the work of the vDocumentation team. It's a lightweight version of the `Get-ESXIODevice.ps1` function.

https://github.com/arielsanchezmora/vDocumentation/blob/master/powershell/vDocumentation/Public/Get-ESXIODevice.ps1

## Related Issue
Related to issue #19.

## How Has This Been Tested?
For now, the code has been tested only with my lab.

## Screenshots (if appropriate):
![image](https://user-images.githubusercontent.com/9823778/46699892-755ccf00-cc1b-11e8-98da-368bcbb1e482.png)

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.

equelin committed:
Add PCI devices drivers and firmwares details for each hosts.

Ideas come from issue #19

tpcarman reviewed (changes requested):
Great work @equelin thank you.

Could I kindly ask you make some minor amendments before we commit.

- Change table header label from `VMKernel Name` to `VMkernel Name`. This is to align with the spelling and casing within the GUI. 
- If possible, would you mind correcting the header labels in the ESXi Host Network table. That's my mistake and saves me having to do another commit. Corrected code is below.

```$VMHostNetwork = $VMhost | Get-VMHostNetwork | Select-Object  VMHost, @{L = 'Virtual Switches'; E = {($_.VirtualSwitch) -join ", "}}, @{L = 'VMkernel Adapters'; E = {($_.VirtualNic) -join ", "}}, @{L = 'Physical Adapters'; E = {($_.PhysicalNic) -join ", "}}, @{L = 'VMkernel Gateway'; E = {$_.VMKernelGateway}}, @{L = 'IPv6 Enabled'; E = {$_.IPv6Enabled}}, @{L = 'VMkernel IPv6 Gateway'; E = {$_.VMKernelV6Gateway}}, @{L = 'DNS Servers'; E = {($_.DnsAddress) -join ", "}}, @{L = 'Host Name'; E = {$_.HostName}}, @{L = 'Domain Name'; E = {$_.DomainName}}, @{L = 'Search Domain'; E = {($_.SearchDomain) -join ", "}}```

- Update Function naming to align with PascalCase, update to `Get-PciDeviceDetails`
- Update output `NA` to `N/A`
- Update Section heading name to `PCI Devices Drivers & Firmware`
- Update Table name to `"$VMhost PCI Devices Drivers & Firmware"`

```#region ESXi Host PCI Devices Details
Section -Style Heading5 'PCI Devices Drivers & Firmware' {
    $VMhostPciDevicesDetails = Get-PciDeviceDetail -Server $vCenter -esxcli $esxcli | Sort-Object 'VMkernel Name'
    $VMhostPciDevicesDetails | Table -Name "$VMhost PCI Devices Drivers & Firmware" 
}
#endregion ESXi Host PCI Devices Details```

equelin committed:
Apply recommendations from @tpcarman

equelin commented:
Hi @tpcarman,

Please look at my last commit, i hope I did what you asked for.

Regards,

tpcarman commented:
Almost there @equelin.

- Change `Get-PcideviceDetail` to `Get-PciDeviceDetail`
- Line 1015 should be `$VMhostPciDevicesDetails = Get-PciDeviceDetail -Server $vCenter -esxcli $esxcli | Sort-Object 'VMkernel Name'`
- Code within lines 1193-1194 also needs updating to correct my VMkernel casing errors
   - `@{L = 'VMkernel Gateway'; E = {$_.VMKernelGateway}}`
   - `@{L = 'VMkernel IPv6 Gateway'; E = {$_.VMKernelV6Gateway}}`

equelin committed:
Correct casing errors.

Change Get-PcideviceDetail to Get-PciDeviceDetail.
Correct VMkernel casing error.

equelin commented:
My bad... That's what happens when you have only a few minutes to code between to meetings... Hope everything is alright now !
